python/pyiceberg/expressions/literals.py

```diff
@@ -125,81 +127,73 @@ def literal(value) -> Literal:
 
 
 @literal.register(bool)
-def _(value: bool) -> Literal[bool]:
+def _(value: bool) -> BooleanLiteral:
```

Comment:
Why require these are a specific literal rather than using the generic `Literal[bool]`?

Reply:
`BooleanLiteral` extends from `Literal[bool]`, so we get the `Literal[bool]` for free. I think it is nice to be specific here.